### PR TITLE
Package Manifest changes to add capabilities

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues

- Related to https://github.com/elastic/integrations/pull/9938

## Summary

- Add security capability to package manifest to match to the changes https://github.com/elastic/integrations/pull/9938
- This should backport to 8.12 as we use to format version 3.0.0 and this capability is available from 2.10.0 https://github.com/elastic/package-spec/blob/db9a76d9534bca28a227d24543d80c5413c77ed6/spec/changelog.yml#L184
- We don't have to worry on packages 8.11 and below as this change is tightly coupled to RegistryPackageManifestV3 Data class.

### CODE REVIEW

File: detection_rules/schemas/registry_package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 @dataclass
 class ConditionElastic:
     subscription: str
+    capabilities: List[str]
```
Comment (terrancedejesus) [inline detection_rules/schemas/registry_package.py:18]: May want to make this optional so it does not break backport builds.

Comment (shashank-elastic) [reply detection_rules/schemas/registry_package.py:18]: for versions 8.11 and below, this wont be applicable as they will use class RegistryPackageManifestV1 was my understanding.

Comment (shashank-elastic) [reply detection_rules/schemas/registry_package.py:18]: the need for making it optional was discussed in DED sync, it was more a build related failures to be avoided during backported.

File: detection_rules/etc/packages.yaml [hunk 2]
```diff
@@ -11,6 +11,8 @@ package:
     conditions:
       elastic:
         subscription: basic
+        capabilities:
+          - security
```
[the remark this answers is not in the document]
Comment (brokensound77) [reply detection_rules/etc/packages.yaml:15]: oh thats right - just saw Terrances comment!

**Comment by brokensound77** [comment]
> Reminder that `packages.yaml` does not backport and thus when merged, we will need to go to 8.12, 8.13, 8.14 branches and add `capabilities:security`.

Is that field backwards compatible across all those stack versions?

**Review by brokensound77 — APPROVED** [review]

File: detection_rules/etc/packages.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (shashank-elastic) [reply detection_rules/etc/packages.yaml:15]: Its made optional so non compatible releases such as 8.11 and below wont have a failure and it is supported to 8.12 and above.

**Review by terrancedejesus — APPROVED** [review]
Yeppers, good work!

**Review by Mikaayenson — APPROVED** [review]
LGTM! Based on our zoom session earlier!

**Pull request merged**